packages/client/src/components/dashboards/cards/status-card.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { referenceForModel } from '@odf/shared/utils';
+import {
+  HealthState,
+  WatchK8sResource,
+  useK8sWatchResource,
+} from '@openshift-console/dynamic-plugin-sdk';
+import {
+  HealthBody,
+  HealthItem,
+} from '@openshift-console/dynamic-plugin-sdk-internal';
+import * as _ from 'lodash-es';
+import {
+  GalleryItem,
+  Gallery,
+  Card,
+  CardHeader,
+  CardTitle,
+} from '@patternfly/react-core';
+import { StorageClientModel } from '../../../models';
+import { StorageClient } from '../../../types';
+
+const clientResource: WatchK8sResource = {
+  kind: referenceForModel(StorageClientModel),
+};
+
+const getStorageClientHealth = (
+  storageClient: StorageClient,
+  loaded: boolean,
+  loadError: any
+) => {
+  if (loadError) {
+    return { state: HealthState.NOT_AVAILABLE, message: '' };
+  }
+  if (!loaded) {
+    return { state: HealthState.LOADING, message: '' };
+  }
+  if (_.isEmpty(storageClient)) {
+    return { state: HealthState.NOT_AVAILABLE };
+  }
+
+  const phase = storageClient.status.phase;
+  switch (phase) {
+    case 'Initializing':
+    case 'Progressing':
+    case 'Onboarding':
+    case 'Offboarding':
+      return { state: HealthState.PROGRESS, message: phase };
+    case 'Connected':
+      return { state: HealthState.OK, message: phase };
+    case 'Failed':
+      return { state: HealthState.ERROR, message: phase };
+  }
+};
+
+export const StatusCard: React.FC = () => {
+  const { t } = useCustomTranslation();
+  const [data, loaded, loadError] =
+    useK8sWatchResource<StorageClient[]>(clientResource);
+
+  const cephHealth = getStorageClientHealth(data[0], loaded, loadError);
```

Comment:
will be always have single `StorageClient` CR, even in case of multiple providers ??
```suggestion
  const cephHealth = getStorageClientHealth(data?.[0], loaded, loadError);
```